src/block/block_read.c

```diff
@@ -221,7 +221,31 @@ __wt_block_read_off(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_ITEM *buf, uin
     check_size = F_ISSET(&swap, WT_BLOCK_DATA_CKSUM) ? size : WT_BLOCK_COMPRESS_SKIP;
     if (swap.checksum == checksum) {
         blk->checksum = 0;
-        if (__wt_checksum_match(buf->mem, check_size, checksum)) {
+
```

Comment:
Haha it happens - thanks for the review!

I have reformatted the changes into a do while loop, so hopefully there isn't any logic missing and less repeat code. The loop will only retry once per bad block - and it will retry on either checksum failures.

This is the gist of the changes, I may need to tidy things up and will do some more in-depth testing next week.